### ISSUE TITLE
Windows 8 Usage: All 3 EFI trickes needed?

### ISSUE THREAD

**Issue by juniordiscart** [0xbb/gpu-switch#5]
Hi,

First of all a big thank you for the effort you put into this. I have a question on whether the tricks needed to get this working on Windows only.
I have a Macbook Pro 11,3 model (mid 2014), but I prefer to work under Windows only (I don't even have OS X or another unix distro installed on my machine). Is it possible to get this to work with just using the first trick in the list (the apple_set_os.efi), or would I also need the last two (patched GRUB and kernel) to get this to work?

If your answer is yes to the question above, can rEFIt be used to place the .efi binary at the correct location on the EFI partition through Windows?

Sorry to post this as an "issue" on your project.

Thanks in advance!

**Comment by 0xbb** [comment]
Hi!
Absolutely no problem for posting this as an issue.

You only need to apply one of the hacks and I also would recommend using [apple_set_os.efi](https://github.com/0xbb/apple_set_os.efi).

Yes, that is exactly the purpose of the efi binary. I haven't tested rEFI but I was successful with rEFInd.
You can install it like described here: http://www.rodsbooks.com/refind/installing.html#windows
Then you can just copy [apple_set_os.efi](https://github.com/0xbb/apple_set_os.efi/releases) to `S:\custom\` and you should be able to access the Intel GPU.

Greetings

**Comment by juniordiscart** [comment]
Alright! Thank you for your reply. I will try this in a few weeks (currently have a few deadlines, so I'm not willing to risk fiddling with the EFI partition and needing to recover the partition tables or do a complete restore of my system if it all goes sideways).

I will also take a look at your link for rEFInd. rEFIt was just the first result I found when looking for accessing the EFI partition on a MacBook.

Thanks!

**Comment by lochotzke** [comment]
Hello,

I'm also trying to get gpu-switch to work on my MacBook Pro 11,3. I have Windows 8.1 64bit installed using BootCamp along side OS X 10.10.3.
I installed rEFInd and placed apple_set_os.efi in EFI/custom on the ESP partition. rEFInd starts at boot and I can select apple_set_os. After a short flicker I then select Windows. It boots (boot animation) but I end up with a pretty much black login screen. Every now and then the shutdown button is shortly visible in white. I managed to install the Intel driver by booting into safe mode. It's still not working though.

What am I missing?

Thanks in advance

**Comment by 0xbb** [comment]
Hello!
I think the black screen is related to some weirdness of the Nvidia driver. At the moment I don't have my Windows hard drive with me and can't test it myself.

Maybe @ah- can help you.

You could also try to get some assistance here:
http://forum.techinferno.com/implementation-guides-apple/8717-%5Bguide%5D-enabling-intel-iris-pro-hd5200-win8-1-15-macbook-pro-gt750m-1.html
or 
https://www.reddit.com/r/apple/comments/354vw8/howto_enabling_the_intel_iris_in_bootcamp_650m750m/

Greetings
Bruno

**Comment by juniordiscart** [comment]
I have been successful in activating the Iris Pro integrated card, but I run into the problem that, when I have both the integrated and dedicated card active, the laptop will boot with a black screen with some flashes of a few login icons (most noticeably the ethernet icon).

I doubt this has anything directly to do with the gpu-switch application, but is there anything that needs to be done specifically to get both working at the same time? I can get them active separately, but it would be nice to have them both working at the same time.

**Comment by ah-** [comment]
@juniordiscart: I tried it again today after updating to Windows 10, and it seems the black screen problem is now gone. Not sure if it was the reinstallation of everything though, but no more issues with that.

**Comment by juniordiscart** [comment]
@ah- I tried to follow the procedure again a few weeks (or maybe a month ago) and got it to work as well: I could have the iGPU and dGPU active at the same time, but I had a constant load of about 10% on the system, from the System process. Did you experience this as well?

**Comment by ah-** [comment]
@juniordiscart yeah, I saw the same thing, not sure what caused that. Haven't investigated that yet.

**Comment by juniordiscart** [comment]
@ah- Just to further add additional details, my previous comment was regarding Windows 8.1. Now I'm running Windows 10, and experiencing the same problems. Some one on Tech Inferno mentioned that putting the Macbook to sleep and then waking it again resolves the problem. I can confirm this as well. Most notably, I also see a large amount of System Interrupts in the processes. I assume it is related to the high System process CPU usage. So the sleep state changes something that causes the System provess to exhibit normal behaviour (I assume).
